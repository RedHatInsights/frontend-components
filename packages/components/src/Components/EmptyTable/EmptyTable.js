import React from 'react';
import propTypes from 'prop-types';

import classNames from 'classnames';

import './EmptyTable.scss';

const EmptyTable = ({ centered, className, children, ...props }) => {

    const emptyTableClasses = classNames(
        'ins-c-table__empty',
        { [`is-centered`]: centered },
        className
    );

    return (
        <div className={ emptyTableClasses } { ...props }> { children }</div>
    );
};

export default EmptyTable;

EmptyTable.propTypes = {
    centered: propTypes.bool,
    children: propTypes.any,
    className: propTypes.string
};
