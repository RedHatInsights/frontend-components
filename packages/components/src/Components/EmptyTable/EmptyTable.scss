@import '~@redhat-cloud-services/frontend-components-utilities/files/Utilities/_all.scss';

.ins-c-table__empty {
    background: var(--pf-global--Color--light-100);
    border-bottom: 1px solid var(--pf-global--BorderColor--light-200);
    @include rem('padding', 20px);
    &.is-centered {
        display: flex;
        justify-content: center;
    }
}
