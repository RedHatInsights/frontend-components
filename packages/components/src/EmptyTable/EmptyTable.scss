@use '~@redhat-cloud-services/frontend-components-utilities/styles/_mixins.scss' as m;

.ins-c-table__empty {
  background: var(--pf-t--global--background--color--primary--default);
  border-bottom: 1px solid var(--pf-t--global--background--color--secondary--default);
  @include m.rem('padding', 20px);
  &.is-centered {
    display: flex;
    justify-content: center;
  }
}
