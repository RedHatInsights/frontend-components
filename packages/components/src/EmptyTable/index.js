export { default } from './EmptyTable';
export { default as EmptyTable } from './EmptyTable.js';
