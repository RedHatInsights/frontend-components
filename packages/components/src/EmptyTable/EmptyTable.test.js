import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EmptyTable from './EmptyTable';

describe('EmptyTable component', () => {
    it('should render', () => {
        const wrapper = shallow(<EmptyTable>Some</EmptyTable>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with centered children', () => {
        const wrapper = shallow(<EmptyTable centered>Centered</EmptyTable>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
