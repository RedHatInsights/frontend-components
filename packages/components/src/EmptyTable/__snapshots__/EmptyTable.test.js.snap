// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyTable component should render 1`] = `
<div
  className="ins-c-table__empty"
>
   
  Some
</div>
`;

exports[`EmptyTable component should render with centered children 1`] = `
<div
  className="ins-c-table__empty is-centered"
>
   
  Centered
</div>
`;
